crates/bevy_ecs/src/world/unsafe_world_cell.rs

```diff
@@ -651,15 +651,14 @@ impl<'w> UnsafeEntityCell<'w> {
     /// - no other mutable references to the component exist at the same time
     #[inline]
     pub unsafe fn get<T: Component>(self) -> Option<&'w T> {
-        let component_id = self.world.components().get_id(TypeId::of::<T>())?;
-
         // SAFETY:
-        // - entity location is valid
-        // - proper world access is promised by caller
+        // - `storage_type` is correct (T component_id + T::STORAGE_TYPE)
+        // - `location` is valid
+        // - proper aliasing is promised by caller
         unsafe {
             get_component(
                 self.world,
-                component_id,
+                self.world.components().get_id(TypeId::of::<T>())?,
```

Comment:
Why was this moved back inside the unsafe block? It's still safe.

Reply:
I thought it was more explicit. It's a bit more obvious that the code is safe, one of the invariants is that "`component_id` should be that of `T`". With this change, we can see `TypeId::of::<T>` _inline_ where it's relevant, rather than somewhere else.

Though of course it's a minor improvement, since the somewhere else in question is not that far off. It wasn't mindless. But I can revert it.